src/Reduced.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ElegantBro\Arrayee;
+
+use ElegantBro\Interfaces\Arrayee;
+use Exception;
+use function array_merge;
+
+final class Reduced implements Arrayee
+{
+    /**
+     * @var Arrayee
+     */
+    private $arrayee;
+    /**
+     * @var callback
+     */
+    private $callback;
+    /**
+     * @var ?Arrayee
+     */
+    private $initial;
+
+    /**
+     * Reduced constructor.
+     * @param Arrayee $arrayee
+     * @param callable $callback
```

Comment:
the description about `$callback` signature will be helpful `function(array $carry, $item): array`

Reply:
@stepanets i don't have knowledge about this hint

I mixed from this article https://phpstan.org/writing-php-code/phpdoc-types

callable(int $foo, string $bar): void (accepts an integer and a string, doesn’t return anything; parameter names are optional and insignificant)

And your comment

```php
    /**
     * Reduced constructor.
     * @param Arrayee $arrayee
     * @param callable(array, mixed): array $callback function(array $carry, $item): array
     * @param Arrayee $initial
     */
```
What do you think about?